dapp/src/utils/orangekit/ledger-live/tests/bitcoin-provider.test.ts

```diff
@@ -0,0 +1,500 @@
+      })
+
+      it("should get the public key from the bitcoin module", () => {
+        expect(mockedWalletApiClient.bitcoin.getPublicKey).toHaveBeenCalledWith(
+          mockedAccount.id,
+          "0/0",
+        )
+      })
+
+      it("should return correct public key", () => {
+        expect(result).toBe(publicKey)
+      })
+    })
+
+    describe("disconnect", () => {
+      beforeAll(async () => {
+        await provider.disconnect()
+      })
+
+      it("should close the connection", () => {
+        expect(mockedTransport.disconnect).toHaveBeenCalled()
+      })
+    })
+  })
+
+  describe("when not connected", () => {
+    beforeAll(() => {
+      provider = new AcreLedgerLiveBitcoinProvider(
+        BitcoinNetwork.Testnet,
+        mockedTransport as unknown as WindowMessageTransport,
+        mockedWalletApiClient as unknown as WalletAPIClient,
+      )
+    })
+
+    describe.each([
+      { methodName: "getPublicKey", args: [] },
+      { methodName: "sendBitcoin", args: [] },
+      { methodName: "signMessage", args: [] },
+      { methodName: "signWithdrawMessage", args: ["", withdrawalData] },
+      { methodName: "getAddress", args: [] },
+      { methodName: "getBalance", args: [] },
+    ])("$methodName", ({ methodName, args }) => {
+      it("should throw an error", async () => {
+        await expect(() =>
+          // @ts-expect-error test
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call
+          provider[methodName](...args),
+        ).rejects.toThrowError("Connect first")
+      })
+    })
+  })
+
+  describe("connect", () => {
+    const account2 = {
+      ...mockedAccount,
+      id: "2",
+      address: "bc1qxg2cc7p5yur7tr3rr2cl5jw47lluuvy7cr2usu",
+    }
+
+    const account3 = {
+      ...mockedAccount,
+      id: "3",
+      address: "tb1qhp6ug4hp3cc84spx3l8sddh8w5ffjujvjycqv5",
+    }
+    const accounts = [mockedAccount, account2, account3]
+
+    let result: string
+
+    describe("when not connected", () => {
+      describe("when `tryConnectToAddress` is not set", () => {
+        describe("when only one account is available", () => {
+          beforeAll(async () => {
+            mockedWalletApiClient.account.list.mockReturnValueOnce([
+              mockedAccount,
+            ])
+
+            provider = new AcreLedgerLiveBitcoinProvider(
+              BitcoinNetwork.Testnet,
+              mockedTransport as unknown as WindowMessageTransport,
+              mockedWalletApiClient as unknown as WalletAPIClient,
+            )
+
+            result = await provider.connect()
+          })
+
+          it("should list accounts", () => {
+            expect(mockedWalletApiClient.account.list).toHaveBeenCalledWith({
+              currencyIds: ["bitcoin_testnet"],
+            })
+          })
+
+          it("should return the first account", () => {
+            expect(result).toBe(mockedAccount.address)
```

Comment:
Shouldn't this test return the address at 0-index?

Reply:
bb447b2